src/components/Articles.jsx

```diff
@@ -1,24 +1,42 @@
 import { useEffect, useState } from "react";
-import { getAllArticles } from "../api";
+import { getAllArticles, getArticlesByTopic } from "../api";
 import ArticleCard from "./ArticleCard";
 import ArticleStyle from "./ArticleStyle";
 import Loading from "./Loading";
+import Topics from "./Topics";
+import { useSearchParams } from "react-router-dom";
 
 export default function Articles() {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [topic, setTopic] = useState("all");
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const queryTopic = searchParams.get("topic");
 
   useEffect(() => {
-    getAllArticles().then(({ data }) => {
-      setArticles(data.articles);
-      setIsLoading(false);
-    });
-  }, []);
+    if (queryTopic) {
+      setTopic(queryTopic);
+    }
+    if (topic !== "all") {
+      getArticlesByTopic(topic).then(({ data }) => {
```

Comment:
if you used the axios params object (look in northcoders notes) you could handle this in one go